src/Components/WeatherDetails/WeatherDetails.js

```diff
@@ -1,67 +1,107 @@
-import React from 'react';
+import React, { Component } from 'react';
 
+import Button from './WeatherShowMoreLessButton';
+import { FadeInTop } from '../../common/Animations/Animations';
+import WeatherAdditionalDetails from '../WeatherDetails/WeatherAdditionalDetails';
 import WeatherIcon from '../WeatherIcon/WeatherIcon';
 
-const WeatherDetails = ({ data }) => {
-  const { cod } = data;
-  const locationNotFound = cod !== 200 ? true : false;
+class WeatherDetails extends Component {
+  state = {
+    showAdditionalDetails: false,
+    active: false,
+  };
 
-  if (locationNotFound) {
-    return (
-      <div className="weather-details">
-        <div className="weather-details__message alert alert-danger">
-          Sorry, we didn't find the location
-        </div>
-      </div>
-    );
+  handleAdditionalDetailsState = () => {
+    this.setState({
+      showAdditionalDetails: !this.state.showAdditionalDetails,
+    });
+  };
+
+  componentDidMount() {
+    this.setState({ active: true });
   }
 
-  const {
-    name,
-    main: { humidity, pressure, temp, temp_max, temp_min },
-    wind: { speed },
-    sys: { country },
-    weather: [{ icon, desctiption }],
-  } = data;
+  render() {
+    const { cod } = this.props.data;
+    const locationNotFound = cod !== 200 ? true : false;
+    const { showAdditionalDetails, active } = this.state;
 
-  return (
-    <div className="weather-details">
-      <div className="weather-details__message alert alert-success">
-        Weather data for the location: <b>{name}</b>
-      </div>
-      <div className="weather-details__container">
-        <div className="weather-details__icon">
-          <WeatherIcon icon={icon} desctiption={desctiption} />
-        </div>
-        <div className="weather-details__data">
-          <div className="weather-details__item">
-            <b>Country:</b> {country}
-          </div>
-          <div className="weather-details__item">
-            <b>Location:</b> {name}
-          </div>
-          <div className="weather-details__item">
-            <b>Temprature:</b> {temp} °C
-          </div>
-          <div className="weather-details__item">
-            <b>Minimum temprature:</b> {temp_min} °C
-          </div>
-          <div className="weather-details__item">
-            <b>Maximum temprature:</b> {temp_max} °C
+    if (locationNotFound) {
```

Comment:
here in render do this:

```
render () {
  const {dat} = this.state;

  return {
    locationFound ?  
      // Success code here
    :
    <LocationNotFound />
  }
}

```

Extract all html responsible for Not Found message to separated component.
And render success HTML on if(locationFound)